src/Infrastructure/LeanCode.ConfigCat/ConfigCatInitializer.cs

```diff
@@ -15,14 +15,17 @@ public ConfigCatInitializer(IConfigCatClient configCatClient)
     [System.Diagnostics.CodeAnalysis.SuppressMessage(
         "?",
         "CA1031:DoNotCatchGeneralExceptionTypes",
-        Justification = "We don't want any exceptions to be propagated."
+        Justification = "We don't want any exceptions to be propagated, as ConfigCat already logs them."
     )]
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
         try
         {
             await configCatClient.GetAllKeysAsync(stoppingToken);
         }
-        catch { }
+        catch
+        {
+            // We don't want any exceptions to be propagated, as ConfigCat already logs them.
```

Comment:
Maybe this nonsense wouldn't be needed if we just did `await configCatClient.GetAllKeysAsync(stoppingToken).ConfigureAwait(SuppressThrowing);` instead…

That said, if I'm reading .NET's code correctly, the `try`/`catch` is redundant anyway.

Reply:
Ad. SuppressThrowing: [it's not that simple](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMBWAbjz1QGYT0iBhIgbzyM5PtIDYSAHCX4BZAIYBLKAAoAlGw5clwMAE82JAJxEA8gAcAgnIB0TAPZQAZhIDmEMAFMDAd0nBp5q7ftPXE4PrAEhYAzsYAyhB6eo4hIQAqABZgZs5SNrIURAC+RADGYsB5idIAogAeeQ56QRZEDuXyrCSYmtIARAYgbATZ7TD15cYiDnFiNg6ZOYpKnCrqzaja+gBCJp7Wdo4ubh4Wmz47/oHBUGGR0bEJyanpU7kFRSUVVTWng00tbe0r3ay9/UGw1GIXGkyy2Rms3mGiWuj0THW+282z87g2KN8bhOoQiURiIKSKTSUAyEPyhWKZUq1VqUA+sNaHSYfwBAwawLGE3uUMhuCUUL4wgAPFJgAA+eFGJpQ5Q3ZxEKAOBUASSgADcxAAbCRwfQOMCFU4vWmnORUfmzTioADswmMADEUgBbABKowgWvcmEyvKhgsw/FQQlQvFFUAl8LWMstVuA8sVyqIas1Or1egNRosJreFnNsq4tqImAtSj5AtjLSDIbDYsl+kRMat1u0oeMAE0JA4tXB85W5cTE6qNdrdfrDXSc3S+82SHaS37cNkgA==).

Reply:
You threw the exception synchronously, before the task was even returned :(

Reply:
Yeah, exactly ;P Except that you don't know if you have a state machine underneath that wraps the `throw` or not, you must examine the code (and expect it does not change).

Reply:
Ah, I see, so even with conversion to base `Task` it's still problematic.

Welp, `await Task.Yield(); await configCatClient...` it is then.

(I don't actually prefer any particular solution but we might as well take advantage of `BackgroundService` doing the suppress for us.)